### DISCUSSION TITLE
fillScreen() did not work as expected and resulted in a visible tear.

### DISCUSSION THREAD

**Discussion by ws2bhhh** [Bodmer/TFT_eSPI#2734] [Q&A - General] [answered]
1. How to reproduce: fillScreen(TFT_BLACK), then fillScreen(TFT_WHITE), a process in which the screen seems to get stuck for a split second, and the position of the stuck seems to be random. Add delay() doesn't help, but makes things worse. Even if the delay() is increased to about 0.5 seconds, the problem still exists.
2. IDE: Platform IO
3. TFT_eSPI library version: 2.5.31
4. board package version: 2.0.7
5. Processor: ESP32-D0WD-V3, 240MHz
6. TFT driver: **ST7735S** (Yes the model number does have an S character in it)
7. Interface type: Parallel

A dumb solution to this problem:

The screen resolution is 128*128, 1.44 inch. So I think ESP32 is fast enough to handle this simple job.
drawPixel() one by one successfully solves the problem. Works fine with out delay().

```
for (int y = 0; y < 128; y++)
  {
    for (int x = 0; x < 128; x++)
    {
      tft.drawPixel(x, y, TFT_WHITE);
    }
  }
for (int y = 0; y < 128; y++)
{
  for (int x = 0; x < 128; x++)
  {
    tft.drawPixel(x, y, TFT_BLACK);
  }
}
```

**Comment by Bodmer** [comment, marked as answer]
This tearing effect is due to the refresh rate of the display driver compared to the draw speed of the processor.

Good to hear you have a solution.
